test/micro/org/openjdk/bench/jdk/incubator/foreign/StrLenTest.java

```diff
@@ -145,6 +151,27 @@ public int panama_strlen_unsafe_trivial() throws Throwable {
         return res;
     }
 
+    @Benchmark
+    public int panama_strlen_memsegmentpool_allocator() throws Throwable {
+        try(ResourceScope scope = ResourceScope.newConfinedScope()) {
+            final var allocator = memorySegmentPool.allocatorForScope(scope);
+            return (int)STRLEN.invokeExact(CLinker.toCString(str, allocator).address());
```

Comment:
This comparison is not 100% identical - here we're calling CLinker.toCString, while the low level benchmark just does a bulk copy. For some reason the latter seems to work much better, and I get better numbers if I do that in this benchmark too.